client/reader/components/quick-post/index.tsx

```diff
@@ -51,6 +51,7 @@ function QuickPost( {
 	receivePosts: ( posts: PostItem[] ) => Promise< void >;
 	successNotice: ( message: string, options: object ) => void;
 } ) {
+	const state = useSelector( ( state ) => state );
```

Comment:
Instead of returning the full state object, we could simply use the getMostRecentlySelectedSiteId selector here. That should make the rerender and update cycle more consistent with our concerns.